File: gitdot-web/packages/gitdot-api/src/endpoint/repository/update-repository.ts
import { z } from "zod";
import { RepositoryResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const UpdateRepositoryRequest = z.object({
  description: z.string().optional(),
});
export type UpdateRepositoryRequest = z.infer<typeof UpdateRepositoryRequest>;

export const UpdateRepositoryResponse = RepositoryResource;
export type UpdateRepositoryResponse = z.infer<typeof UpdateRepositoryResponse>;

export const UpdateRepository = {
  path: "/repository/{owner}/{repo}",
  method: "PATCH",
  request: UpdateRepositoryRequest,
  response: UpdateRepositoryResponse,
} as const satisfies Endpoint;
export type UpdateRepository = typeof UpdateRepository;

File: gitdot-api/src/endpoint/repository/update_repository.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::RepositoryResource};

pub struct UpdateRepository;

impl Endpoint for UpdateRepository {
    const PATH: &'static str = "/repository/{owner}/{repo}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateRepositoryRequest;
    type Response = UpdateRepositoryResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UpdateRepositoryRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
}

pub type UpdateRepositoryResponse = RepositoryResource;
